frontend/src/components/DeviceList/DeviceList.tsx

```diff
@@ -1,30 +1,40 @@
 import React from 'react'
+import clsx from 'clsx'
 import { useSelector } from 'react-redux'
 import { DeviceListItem } from '../DeviceListItem'
 import { getAccountId, getOwnDevices } from '../../models/accounts'
 import { DeviceSetupItem } from '../DeviceSetupItem'
 import { ApplicationState } from '../../store'
 import { ServiceContextualMenu } from '../ServiceContextualMenu'
 import { Notice } from '../Notice'
-import { List, Divider, ListItem } from '@material-ui/core'
+import { List, Divider, ListItem, makeStyles, createStyles, Theme, Typography } from '@material-ui/core'
 import { LoadMore } from '../LoadMore'
-
+import { FilterDrawerContent } from '../FilterDrawerContent'
+import { Body } from '../Body'
 export interface DeviceListProps {
   devices?: IDevice[]
   connections: { [deviceID: string]: IConnection[] }
+  handleDrawer: (state: boolean) => void
+  open: boolean
 }
 
-export const DeviceList: React.FC<DeviceListProps> = ({ devices = [], connections = {} }) => {
+export const DeviceList: React.FC<DeviceListProps> = ({ devices = [], connections = {}, handleDrawer, open }) => {
   const [contextMenu, setContextMenu] = React.useState<IContextMenu>({})
   const { myDevice, loggedInUser, registeredId } = useSelector((state: ApplicationState) => ({
     registeredId: state.backend.device.uid,
     loggedInUser: getAccountId(state) === state.auth.user?.id,
     myDevice: getOwnDevices(state).find(device => device.id === state.backend.device.uid),
   }))
 
+  const css = useStyles()
+
   return (
-    <>
-      <List>
+    <div className={css.root}>
+      <List
+        className={clsx(css.content, {
+          [css.contentShift]: open,
```

Comment:
We need to replace the <Drawer> MuiComponent with probably a <Collapse> one. Then if in a Flexbox it should all just naturally move over.